src/details/ast.cpp

```diff
@@ -97,7 +98,8 @@ auto Program::body() -> Body { return Body(program.named_children()); }
 };*/
 // BinaryOperation
 BinaryOperation::BinaryOperation(ts::Node node) : bin_op(node) {
-    if (node.type_id() != ts::NODE_BINARY_OPERATION || node.child_count() != 3) {
+    assert(node.child_count() == 3);
```

Comment:
the asserts should be after the if because if we pass in the wrong type of node the exception is more appropriate.